lib/structuraid_core/design_codes/nsr_10/rc/footings/min_height.rb

```diff
@@ -0,0 +1,38 @@
+module StructuraidCore
+  module DesignCodes
+    module NSR10
+      module RC
+        module Footings
+          class MinHeight
+            include DesignCodes::Utils::CodeRequirement
+            use_schema DesignCodes::Schemas::RC::Footings::MinHeightSchema
+
+            MIN_HEIGHT_MAPPINGS = {
+              over_soil: 150,
+              over_piles: 300
+            }.freeze
```

Comment:
ACI only has a minimum effective height that must be fulfilled in any scenario. NSR has two different minimums, depending on whether the footing is positioned over soil directly or over piles